tool/speed.cc

```diff
@@ -1029,8 +1029,25 @@ static bool SpeedECDSACurve(const std::string &name, int nid,
   }
 
   BM_NAMESPACE::UniquePtr<EC_KEY> key(EC_KEY_new_by_curve_name(nid));
-  if (!key ||
-      !EC_KEY_generate_key(key.get())) {
+  if(!key) {
+    return false;
+  }
+
+  TimeResults results;
+  if (!TimeFunction(&results, [&key]() -> bool {
+// We run extra checks on the key in FIPS mode
+#if defined(BORINGSSL_FIPS)
+        return EC_KEY_generate_key_fips(key.get()) == 1;
+#else
+        return EC_KEY_generate_key(key.get()) == 1;
+#endif
+      })) {
+    return false;
+  }
+
+  results.Print(name + " key generation");
```

Comment:
What is the final name in the results? Is it "ECDSA {curve} key generation" instead of just "{curve} key generation"? If so should this be in a separate function that just does key generation since ECDH and ECDSA use the same keygen and this performance is not tied to ECDSA specifically?

Reply:
```
 Did 177557 ECDSA P-256 key generation operations in 20000036us (8877.8 ops/sec)
```